### PR TITLE
Arwa Alkhathlan - Selection-Project

### PR DESCRIPTION
**Pull request by arwaalkhathlan**
a task app using dart

### CODE REVIEW

File: src/core/note_app.dart [hunk 1]
```diff
@@ -1,6 +1,109 @@
...
+
+          NoteModel? taskToEdit =
+              tasks.firstWhereOrNull((task) => task.id == taskId);
+
+          if (taskToEdit != null) {
+            print('What would you like your new task name to be?');
+            taskToEdit.task_name = stdin.readLineSync()!;
+
+            print('Enter the new task description:');
+            taskToEdit.description = stdin.readLineSync()!;
```
Comment (Radwan-Albahrani) [inline src/core/note_app.dart]: add checking if title or description is null
```suggestion
            print('What would you like your new task name to be?');
            String name = stdin.readLineSync()!;

            if(name.isEmpty)
            {
                print("Task name cannot be empty");
                break;
            }
            taskToEdit.task_name = name;



            print('Enter the new task description:');
            taskToEdit.description = stdin.readLineSync()!;
```

File: src/core/note_app.dart [hunk 2]
```diff
@@ -1,6 +1,109 @@
...
+            print(task);
+          }
+          print("♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡");
+          break;
+
+        case 5:
+          print('Exit');
+          print("♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡");
+
+          break;
```
Comment (Radwan-Albahrani) [inline src/core/note_app.dart]: exit the program
```suggestion
          print('Exit');
          print("♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡♡");
          exit(1);
          break;
```

**Review by Radwan-Albahrani — CHANGES_REQUESTED** [review]
Hello! Fantastic work and robust code. While this is a great project, this does not fulfill the requirements of the project. The project needs to be a note taking app, not a TaskList app. Please review the [project requirements](https://github.com/GDSC-IAU/Selection-Project#project-overview) and try again.

**Comment by arwaalkhathlan** [comment]
> Hello! Fantastic work and robust code. While this is a great project, this does not fulfill the requirements of the project. The project needs to be a note taking app, not a TaskList app. Please review the [project requirements](https://github.com/GDSC-IAU/Selection-Project#project-overview) and try again.

i'm sorry i just assumed it's like task 2 i will fix it today so sorry again

**Pull request closed without merging**